### PR TITLE
Configurable Default Repository URLs for Ruby Gems and Python Packages

### PR DESCRIPTION
**Pull request by stephen-ritter**
This is a small patch to allow for the default URLs to be set based on the associated environment variables. `--trusted-host` was also added to account for repositories that are self hosted on local deployments.

### CODE REVIEW

File: openc3-cosmos-init/plugins/packages/openc3-tool-common/src/tools/admin/tabs/settings/PypiSettings.vue [hunk 1]
```diff
@@ -51,7 +51,7 @@ export default {
   mixins: [Settings],
   data() {
     return {
-      pypiUrl: 'https://pypi.org',
+      pypiUrl: process.env.PYPI_URL,
```
Comment (ryanmelt) [inline openc3-cosmos-init/plugins/packages/openc3-tool-common/src/tools/admin/tabs/settings/PypiSettings.vue:54]: Client side web browser code doesn't have environment variables.   This is nodejs style backend code that won't work in the browser. This is only setting the default value and should get overridden with actual value if it has been set.

File: openc3/lib/openc3/models/plugin_model.rb [hunk 2]
```diff
@@ -200,7 +200,7 @@ def self.install_phase2(plugin_hash, scope:, gem_file_path: nil, validate_only:
           end
           unless validate_only
             Logger.info "Installing python packages from requirements.txt with pypi_url=#{pypi_url}"
-            puts `/openc3/bin/pipinstall --no-warn-script-location -i #{pypi_url} -r #{File.join(gem_path, 'requirements.txt')}`
+            puts `/openc3/bin/pipinstall --no-warn-script-location -i #{pypi_url} --trusted-host #{URI.parse(pypi_url).host} -r #{File.join(gem_path, 'requirements.txt')}`
```
Comment (ryanmelt) [inline openc3/lib/openc3/models/plugin_model.rb:203]: I like having the option to set trusted-host but it shouldn't always be on.  That defeats SSL even when you have a good SSL connection.  Maybe an ENV to enable trusted host?

File: openc3/lib/openc3/models/scope_model.rb [hunk 3]
```diff
@@ -379,9 +379,9 @@ def seed_database
       setting = SettingModel.get(name: 'source_url')
       SettingModel.set({ name: 'source_url', data: 'https://github.com/OpenC3/cosmos' }, scope: @scope) unless setting
       setting = SettingModel.get(name: 'rubygems_url')
-      SettingModel.set({ name: 'rubygems_url', data: 'https://rubygems.org' }, scope: @scope) unless setting
+      SettingModel.set({ name: 'rubygems_url', data: ENV['RUBYGEMS_URL'] }, scope: @scope) unless setting
```
Comment (ryanmelt) [inline openc3/lib/openc3/models/scope_model.rb:382]: These should have a default if the ENV isn't set.

File: openc3/lib/openc3/models/scope_model.rb [hunk 4]
```diff
@@ -379,9 +379,9 @@ def seed_database
...
       setting = SettingModel.get(name: 'pypi_url')
-      SettingModel.set({ name: 'pypi_url', data: 'https://pypi.org' }, scope: @scope) unless setting
+      SettingModel.set({ name: 'pypi_url', data: ENV['PYPI_URL'] }, scope: @scope) unless setting
```
Comment (ryanmelt) [inline openc3/lib/openc3/models/scope_model.rb:384]: These should have a default if the ENV isn't set.

File: openc3/lib/openc3/models/python_package_model.rb [hunk 5]
```diff
@@ -92,7 +92,7 @@ def self.install(name_or_path, scope:)
         end
       end
       Logger.info "Installing python package: #{name_or_path}"
-      result = OpenC3::ProcessManager.instance.spawn(["/openc3/bin/pipinstall", "--no-warn-script-location", "-i", pypi_url, package_file_path], "package_install", package_filename, Time.now + 3600.0, scope: scope)
+      result = OpenC3::ProcessManager.instance.spawn(["/openc3/bin/pipinstall", "--no-warn-script-location", "-i", pypi_url, "--trusted-host", URI.parse(pypi_url).host, package_file_path], "package_install", package_filename, Time.now + 3600.0, scope: scope)
```
Comment (ryanmelt) [inline openc3/lib/openc3/models/python_package_model.rb:95]: I like having the option to set trusted-host but it shouldn't always be on. That defeats SSL even when you have a good SSL connection.  Maybe add an ENV to enable trusted host?

File: openc3-cosmos-init/plugins/packages/openc3-tool-common/src/tools/admin/tabs/settings/RubyGemsSettings.vue [hunk 6]
```diff
@@ -55,7 +55,7 @@ export default {
   mixins: [Settings],
   data() {
     return {
-      rubygemsUrl: 'https://rubygems.org',
+      rubygemsUrl: process.env.RUBYGEMS_URL,
```
Comment (ryanmelt) [inline openc3-cosmos-init/plugins/packages/openc3-tool-common/src/tools/admin/tabs/settings/RubyGemsSettings.vue:58]: Client side web browser code doesn't have environment variables. This is nodejs style backend code that won't work in the browser.  This is only setting the default value and should get overridden with actual value if it has been set.

**Review by ryanmelt — CHANGES_REQUESTED** [review]

File: openc3-cosmos-init/plugins/packages/openc3-tool-common/src/tools/admin/tabs/settings/PypiSettings.vue (hunk 1 above)
Comment (stephen-ritter) [reply openc3-cosmos-init/plugins/packages/openc3-tool-common/src/tools/admin/tabs/settings/PypiSettings.vue]: Ah, that makes sense. I reverted this back to what it was.

File: openc3/lib/openc3/models/plugin_model.rb (hunk 2 above)
Comment (stephen-ritter) [reply openc3/lib/openc3/models/plugin_model.rb]: I added an environment variable called `PIP_ENABLE_TRUSTED_HOST` and only apply `--trusted-host` if that environment variable is set.

File: openc3/lib/openc3/models/scope_model.rb (hunk 3 above)
Comment (stephen-ritter) [reply openc3/lib/openc3/models/scope_model.rb]: I adjusted this so now the default is `https://rubygems.org` if `ENV['RUBYGEMS_URL']` is not set.

File: openc3/lib/openc3/models/scope_model.rb (hunk 4 above)
Comment (stephen-ritter) [reply openc3/lib/openc3/models/scope_model.rb]: I adjusted this so now the default is `https://pypi.org` if `ENV['PYPI_URL']` is not set.

File: openc3/lib/openc3/models/python_package_model.rb (hunk 5 above)
Comment (stephen-ritter) [reply openc3/lib/openc3/models/python_package_model.rb]: I added an environment variable called `PIP_ENABLE_TRUSTED_HOST` and only apply `--trusted-host` if that environment variable is set.

File: openc3-cosmos-init/plugins/packages/openc3-tool-common/src/tools/admin/tabs/settings/RubyGemsSettings.vue (hunk 6 above)
Comment (stephen-ritter) [reply openc3-cosmos-init/plugins/packages/openc3-tool-common/src/tools/admin/tabs/settings/RubyGemsSettings.vue]: Reverted this back.

**Review by ryanmelt — APPROVED** [review]

**Pull request merged**